source/common/upstream/eds.h

```diff
@@ -78,7 +78,9 @@ class EdsClusterImpl
   const LocalInfo::LocalInfo& local_info_;
   const std::string cluster_name_;
   std::vector<LocalityWeightsMap> locality_weights_map_;
-  HostMap all_hosts_;
+
+  HostMapSharedPtr all_host_map_;
```

Comment:
Instead of having this be in the EDS cluster only, I would recommend moving this into the cluster base class, or directly in the priority set. This way it will be used and populated by all cluster types (per my other comment). Additionally, if we do this, I think we can remove all of the special case code you added in cluster manager impl. It's just more data that should be updated and posted on updates. I don't fully see the value of all that new code.

Reply:
@mattklein123 
I added the `postThreadLocalHostMapUpdate` method to the `clusterManager` to ensure that `all_host_map_ in` the EDS Cluster can be synchronized to each worker thread after the update. When `batchUpdate` is used, `postThreadLocalClusterUpdate` cannot do this.
If you maintain all_host_map_ directly in PrioritySet, it is indeed possible to remove these codes. But it may bring some frequent HostMap copy problems (I'm not entirely sure).
But I agree that we can implement a general all_host_map first, and then continue. So we can continue to discuss this issue in the new PR.

Reply:
IMO it won't be too costly to maintain a map in the priority set and then update it when we update everything else. Let's try that first and optimize later if needed?

Reply:
Okay, I will prepare a new PR as soon as possible this week. 👌

Reply:
@mattklein123 https://github.com/envoyproxy/envoy/pull/17290